src/speed-index.js

```diff
@@ -114,9 +114,12 @@ function calculateSpeedIndexes(frames, data) {
 	let perceptualSpeedIndex = firstPaintTs - startTs;
 
 	frames.forEach(function (frame) {
-		const elapsed = frame.getTimeStamp() - prevFrameTs;
-		speedIndex += elapsed * (1 - prevProgress);
-		perceptualSpeedIndex += elapsed * (1 - prevPerceptualProgress);
+		// skip frames from 0 to fP
+		if (frame.getTimeStamp() > firstPaintTs) {
+			const elapsed = frame.getTimeStamp() - prevFrameTs;
```

Comment:
I think I remember reading that in v8 declaring a const or let inside of a loop is a major perf hit ATM.
not a big deal. 🐐